consensus/polybft/consensus_runtime.go

```diff
@@ -354,6 +354,40 @@ func (c *consensusRuntime) FSM() error {
 	isEndOfSprint := c.isEndOfSprint(pendingBlockNumber)
 	isEndOfEpoch := c.isEndOfEpoch(pendingBlockNumber)
 
+	valSet, err := NewValidatorSet(epoch.Validators)
+	if err != nil {
+		return fmt.Errorf("cannot create validator set for fsm: %w", err)
+	}
+
+	if epoch.Number > 0 { // it must be at least epoc
```

Comment:
I don't think we need this condition. First epoch will always be epoch 1 when we start the chain (as per smart contracts).